interactors/nonceHandlerV3/workers/transactionWorker.go

```diff
@@ -104,35 +105,39 @@ func (tw *TransactionWorker) AddTransaction(transaction *transaction.FrontendTra
 		return r
 	}
 
-	tw.responsesChannels[transaction.Nonce] = r
+	// check if a tx with the same nonce is currently being sent.
+	if _, ok := tw.responsesChannels[transaction.Nonce]; ok {
+		r <- &TransactionResponse{TxHash: "", Error: fmt.Errorf("transaction with nonce:"+
+			" %d has already been scheduled to send", transaction.Nonce)}
+		return r
+	}
 
+	tw.responsesChannels[transaction.Nonce] = r
 	heap.Push(&tw.tq, &TransactionQueueItem{tx: transaction})
 	return r
 }
 
 // start will spawn a goroutine tasked with iterating all the transactions inside the priority queue. The priority is
 // given by the nonce, meaning that transaction with lower nonce will be sent first.
-func (tw *TransactionWorker) start(ctx context.Context, pollingInterval time.Duration) {
-	ticker := time.NewTicker(pollingInterval)
+// All these transactions are send with an interval between them.
+func (tw *TransactionWorker) start(ctx context.Context, intervalToSend time.Duration) {
+	ticker := time.NewTicker(intervalToSend)
 
 	go func() {
-		chMoreWorkToDo := make(chan struct{}, 1)
 		for {
 			select {
 			case <-ctx.Done():
 				log.Info("context cancelled - transaction worker has stopped")
 				tw.closeAllChannels(ctx)
 				return
 			case <-ticker.C:
-				tw.processNextTransaction(ctx, chMoreWorkToDo)
-			case <-chMoreWorkToDo:
-				tw.processNextTransaction(ctx, chMoreWorkToDo)
+				tw.processNextTransaction(ctx)
```

Comment:
Is this change desired? Now the component will send one transaction at the interval time. In the previous variant it would have sent pending transactions one after another

Reply:
the proxy can't handle that many requests anyway. This way you can leverage the `intervalToSend` in order to be sure that 100% of the transactions sent will be executed.

Reply:
We can not make assumptions that the proxy can not handle a lot of requests. The public proxy might have some antiflood protections but that might not be the case on some custom setups where this service might be installed. You can rework this code in 2 different ways:
1. you can add a different ticker with a different interval for consecutive sends, or
2. can use the proxy.SendTransactions to send a batch of transactions.

"Putting delays" usually solves some problems but degrades the performance

Reply:
I could decrease the minimum limit. One could put 1 millisecond interval  instead and still retain flexibility.